src/mixins/numeral-formatable.js

```diff
@@ -0,0 +1,267 @@
+      this.sign = '+'
+      return this.addPrefixSuffix(zero.toFixed(this.precision).toString().replace('.', this.decimal))
+    },
+    oneChar (char, defaultChar = '.') {
+      return char ? char[0] : defaultChar
+    },
+    // TODO: Rework this restrictive function to
+    // make -ve as a valid value in order
+    // to support special meaning/feature
+    absVal (value) {
+      const parsed = parseInt(value, 10)
+
+      if (isNaN(parsed)) return 0
+      return Math.abs(parsed)
+    },
+    adjustNumeralCaret (selection, text) {
+      // Caret is at prefix
+      if (selection === 0) return this.numeralPrefix.length + this.sign.length
+
+      // Whether selection !== 0 (may happen when block size === 0) or it's an
+      // integer or a fractional where abs(value) >= 10
+      if (selection !== text.length && (!this.precision ||
+        (String(this.lazyValue).length > this.precision + (this.sign ? 3 : 2)))) {
+        return selection
+      }
+
+      // Caret is at suffix
+      const suffixPosition = this.numeralSuffix ? text.lastIndexOf(this.numeralSuffix) : 0
+      if (suffixPosition > 0 && selection >= suffixPosition) {
+        return selection > suffixPosition ? suffixPosition
+          : this.delete || this.backspace ? suffixPosition : suffixPosition - 1
+      }
+
+      // Handle abs(value) < 10
+      // Make the caret stay where it were when the
+      // length of input is the same before and after
+      return this.backspace ? this.numeralCaretOnBackspace(selection, text)
+        : this.delete ? this.numeralCaretOnDelete(selection, text)
+          : this.numeralCaretOnInsert(selection)
+    },
+    numeralCaretOnInsert (selection) {
+      selection -= this.selection < selection ? 1 : 0
+      selection -= this.selection === selection ? 1 : 0
+
+      return selection
+    },
+    numeralCaretOnDelete (selection, text) {
+      if (text.substr(this.selection, 1) === this.decimal) {
+        selection += 1
+      } else {
+        selection += text.substr(this.numeralPrefix.length + this.sign.length, 1) === '0' ? 1 : 0
+        selection += text.substr(selection - 1, 1) === this.decimal ? 1 : 0
+      }
+
+      return selection
+    },
+    numeralCaretOnBackspace (selection, text) {
+      if (text.substr(this.numeralPrefix.length + this.sign.length, 1) === '0') {
+        // abs(value) < 1
+        text.substr(this.selection, 1) !== this.decimal && selection++
+        text.substr(this.selection - 1, 1) === this.decimal && selection++
+      }
+
+      return selection
+    },
+    // Correction on arbitrary value entered externally
+    attemptNumeralCorrection (text) {
+      const decimalPoint = text.lastIndexOf('.')
+      let integer = decimalPoint < 0 ? this.removeNonNumeral(text)
+        : this.removeNonNumeral(text.substring(0, decimalPoint))
+
+      integer = integer.length ? integer : '0'
+      if (!this.precision) return integer
+
+      return decimalPoint < 0 ? integer + this.getFraction('')
```

Comment:
similar to `parseFloat ()` below I think you can also use here built in `toFixed()` function
Probably
```js
return Number(this.removeNonNumeral(text)).toFixed(this.precision)
```
or something similar (and after this change and change in `parseFloat()` `getFraction` could be removed)

Reply:
My rationale above.